clar/tests/test_solvers.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import pytest
+
+from clar.solvers import solver, get_path
+from clar.utils import get_alpha_max, get_sigma_min
+from clar.data.artificial import get_data_me
+
+
+def test_mtl():
+    rho_noise = 0.3
+    SNR = 1
+    n_epochs, n_channels, n_sources, n_times = 5, 20, 10, 30
+    pb_name = "MTL"
+    tol = 1e-10
+
+    X, all_epochs, _, _ = get_data_me(
+        dictionary_type="Gaussian", noise_type="Gaussian_iid",
+        n_epochs=n_epochs, n_channels=n_channels, n_times=n_times,
+        n_sources=n_sources, n_active=3, rho_noise=rho_noise,
+        SNR=SNR)
+
+    Y = np.mean(all_epochs, axis=0)
+    sigma_min = get_sigma_min(Y)
+    alpha_max = get_alpha_max(X, Y, sigma_min, pb_name=pb_name)
+
+    alpha_div = 5
+    alpha = alpha_max / alpha_div
+
+    B_mtl, _, E, gaps = solver(
+        X, Y, alpha, sigma_min, B0=None,
+        tol=tol, pb_name=pb_name, n_iter=10000)
+    np.testing.assert_array_less(gaps[-1], tol)
+
+    _, _, E, gaps = solver(
+        X, Y, alpha, sigma_min, B0=B_mtl,
+        tol=tol, pb_name=pb_name, n_iter=10000)
+    np.testing.assert_equal(len(E), 2)
```

Comment:
is it normal that convergence happens so fast?

Reply:
it seems fair since we are passing B0=B_mtl with the same arguments ?